src/pages/NewChatPage.js

```diff
@@ -0,0 +1,180 @@
+import React, {Component} from 'react';
+import {View} from 'react-native';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import OptionsSelector from '../components/OptionsSelector';
+import {getNewChatOptions} from '../libs/OptionsListUtils';
+import ONYXKEYS from '../ONYXKEYS';
+import styles from '../styles/styles';
+import {fetchOrCreateChatReport} from '../libs/actions/Report';
+import KeyboardSpacer from '../components/KeyboardSpacer';
+import withWindowDimensions, {windowDimensionsPropTypes} from '../components/withWindowDimensions';
+import {hide as hideSidebar} from '../libs/actions/Sidebar';
+
+const personalDetailsPropTypes = PropTypes.shape({
+    // The login of the person (either email or phone number)
+    login: PropTypes.string.isRequired,
+
+    // The URL of the person's avatar (there should already be a default avatarURL if
+    // the person doesn't have their own avatar uploaded yet)
+    avatarURL: PropTypes.string.isRequired,
+
+    // This is either the user's full name, or their login if full name is an empty string
+    displayName: PropTypes.string.isRequired,
+});
+
+const propTypes = {
+    // All of the personal details for everyone
+    personalDetails: PropTypes.objectOf(personalDetailsPropTypes).isRequired,
+
+    // All reports shared with the user
+    reports: PropTypes.shape({
+        reportID: PropTypes.number,
+        reportName: PropTypes.string,
+    }).isRequired,
+
+    // Session of currently logged in user
+    session: PropTypes.shape({
+        email: PropTypes.string.isRequired,
+    }).isRequired,
+
+    ...windowDimensionsPropTypes,
+};
+
+class NewChatPage extends Component {
+    constructor(props) {
+        super(props);
+
+        this.createNewChat = this.createNewChat.bind(this);
+
+        const {
+            personalDetails,
+            userToInvite,
+        } = getNewChatOptions(
+            props.reports,
+            props.personalDetails,
+            '',
+        );
+
+        this.state = {
+            searchValue: '',
+            personalDetails,
+            userToInvite,
+        };
+    }
+
+    /**
+     * Helper method that returns the text to be used for the header's message and title (if any)
+     *
+     * @return {String}
+     */
+    getHeaderTitleAndMessage() {
+        const hasSelectableOptions = this.state.personalDetails.length !== 0;
+        if (!hasSelectableOptions && !this.state.userToInvite) {
+            return {
+                headerTitle: '',
+                // eslint-disable-next-line max-len
+                headerMessage: 'Don\'t see who you\'re looking for? Type their email or phone number to invite them to chat.',
```

Comment:
Maybe we can make the `headerMessage` a constant somewhere and then reference the same text in both NewGroup and NewChat component. This way if we need to change the message we can do it in just one place?